richtextfx/src/main/java/org/fxmisc/richtext/GenericStyledAreaBehavior.java

```diff
@@ -270,10 +270,15 @@
     private final GenericStyledArea<?, ?, ?> view;
 
     /**
-     * Indicates whether selection is being dragged by the user.
+     * Indicates whether an existing selection is being dragged by the user.
      */
     private DragState dragSelection = DragState.NO_DRAG;
 
+    /**
+     * Indicates whether a new selection is being made by the user.
+     */
+    private DragState dragNewSelection = DragState.NO_DRAG;
+    
```

Comment:
Could you explain why this is being added? If I recall correctly, the dragging behavior already works without adding a second one to track things.

Reply:
So this is being added to fix `onNewSelectionDragEnd/Finished`, which wasn't being referenced as mentioned above. This isn't for the dragging of an existing selection, but rather for what happens at the end of creating a new selection via: click on unselected text, drag, release click sequence.

Reply:
I looked through the repo's current code (not the changes you've submitted) and have concluded that we don't have a `onNewSelectionFinished` mouse hook. We have `onNewSelectionDrag`, which is similar but not the same thing. When a user presses the mouse outside of a selection, and then drags the mouse somewhere _but does not release it_, the `newSelectionDrag` hook gets called each time the mouse moves. However, once the mouse is released, there isn't a hook to indicate that the selection will not continue being updated. Hence, your support for this makes sense.

Initially, your approach for adding support for the `onNewSelectionFinished` via another `DragState` variable seemed unnecessary if one could just add a `if (view.getSelection().getLength() != 0) { /* run hook */ }` to the `NO_DRAG` case in the `processMouseReleased()` method. No need to take up additional memory when we already have another way of figuring that out.

However, when I looked through more of the code, I realized that there are some assumptions I am making that would lead to the code breaking if one overrode the mouse behavior code in a specific way. For example, on the first primary mouse button press, I assume that the developer will not override the `onOutsideSelectionMousePress` code, which clears any selection that may have existed before the mouse was pressed. Thus, when the user releases the mouse, no additional code needs to run (e.g. clear any selection) because of my assumption. Should someone override that part of the code, so that it doesn't clear the selection, my approach to adding such support would no longer work whereas yours still would.

All in all, it's tricky to design a default overrideable mouse behavior implementation because of its complexity (e.g. autoscrolling; whether there will be a selection at some point in the code's execution or not; etc.) and the lack of knowledge we have about how it will be overridden.